### DISCUSSION TITLE
potential play-grpc bug with HTTPEntity$Strict

### DISCUSSION THREAD

**Discussion by cregg-yopp** [playframework/playframework#11591] [Q&A] [answered]
Hi all. We've been running into a strange issue using web grpc to communicate with one of our servers using grpc-play. The issue we are seeing is documented thoroughly in the following [repository](https://bitbucket.org/dkichler/play-grpc-sandbox/src/play-action-routers-issue/).

The repo currently contains a, albeit, slightly hacky fix to the issue that we are seeing. However we aren't entirely sure if this issue is by design or is truly a defect / missed case. Should this actually be a bug we'd love to contribute our fix. 

Version Info
Play 2.8.18
play-grpc-generators 0.9.1 
Java 11

All other dependencies can be found in the build.sbt listed above. 

Thanks! Please let me know if there is any more info that I can provide.

**Comment by cregg-yopp** [comment, marked as answer]
Incase it was not obvious from my initial discussion the repo that contains the potential bug is [play-grpc](https://github.com/playframework/play-grpc)

**Reply by mkurz** [reply]
@cregg-yopp So, probably the fix you mention in your bitbucket repo should work, how about opening a pull request in the [play-grpc](https://github.com/playframework/play-grpc) repo? You can target the `0.9.x` branch there, build the project locally and try if that works for you and open a pull request afterwards.

**Reply by cregg-yopp** [reply]
Excellent! Will look to do that in the near future.
